crates/wasm-smith/src/core/code_builder.rs

```diff
@@ -1239,7 +1239,11 @@ impl CodeBuilder<'_> {
         module: &Module,
     ) -> Result<Vec<Instruction>> {
         let max_instructions = module.config.max_instructions;
-        let allowed_instructions = module.config.allowed_instructions;
+        let allowed_instructions = if module.config.allow_floats {
+            module.config.allowed_instructions
+        } else {
+            module.config.allowed_instructions.without_floats()
+        };
```

Comment:
@alexcrichton If floats aren't allowed, we automatically replace each `InstructionKind::Foo`  with `InstructionKind::FooNonFloat`.